include/SH3/system/msg/msg.hpp

```diff
@@ -0,0 +1,73 @@
+/** @file
+ *  Data structure representing a SILENT HILL 3 message.
+ *  These messages (and their attributes) can later be passed off
+ *  to the renderer to be drawn as text strings with FTGL.
+ *  The font used by SILENT HILL 3 is VERY similar to Times New Roman, however as of now,
+ *  we currently don't know where the font actually resides (it's not in the MFT).
+ *
+ *  How this (hopefully) will work:
+ *  The string to print is printed character (slow? Shouldn't be imho).
+ *  During each time we draw, we check whether or not we need to change
+ *  the attribute that we are drawing with (effect, colour etc). The current attribute
+ *  to use is determined by scanning the current character we want to print for 0xFF.
+ *  This character is known as the @ref SH3_ATTR_START character, and signals to the message
+ *  manager that the next character read will be an attribute byte.
+ *
+ *  On entering a 'scene' (level\area), all messages are loaded. by the message manager
+ *
+ *  @note Most, if not all strings in SILENT HILL 3 are centered.
+ *
+ *  @copyright 2016-2017  Palm Studios
+ *
+ *  @date 12-1-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef MSG_HPP_INCLUDED
+#define MSG_HPP_INCLUDED
+
+#include <cstdint>
+
+#define SH3_2_ASCII(x) ((x) - 32) /**< Turns a SILENT HILL 3 msg character into an ASCII character */
```

Comment:
Okay sure thing. The waste I was referring to was a few clock cycles, but I suppose they'd both get compiled into the same thing. 

It's probably a good idea to have it as a function so we at least have the debugging symbol in case something does screw up (and I'm willing to bet something will [laughs]).